### PR TITLE
[3437] Migrate to MUI 5

### LINKED ISSUE #3437: Migrate to MUI 5
The pull request: https://github.com/eclipse-sirius/sirius-web/pull/3438

I suggest updating MUI to version 5 as Material UI 4 is outdated and does not receive security or performance updates. Also it's a complicated task to combine different versions of MUI in a single application. We managed it in our MUI 5 application based on Sirius Web, but it is really painful.

Also it will allow one to migrate to React 18 later.

# What is changed

1) The following dependencies are updated:

```diff
-"@material-ui/core": "4.12.4",
-"@material-ui/icons": "4.11.3",
-"@material-ui/lab": "4.0.0-alpha.61",
+"@mui/material": "5.15.19",
+"@mui/icons-material": "5.15.19",
+"@mui/x-tree-view": "7.6.1",
```

2) Changed namespaces:

```
@material-ui/core -> @mui/material
@material-ui/icons -> @mui/icons-material
```

3) `<TreeView>` component moved to a separate project:

```diff
-import TreeView from '@material-ui/lab/TreeView';
-import { TreeItem } from '@material-ui/lab';
+import { SimpleTreeView } from '@mui/x-tree-view/SimpleTreeView';
+import { TreeItem } from '@mui/x-tree-view/TreeItem';
```

4) Changed `<Autocomplete>` component:

```diff
-import Autocomplete from '@material-ui/lab/Autocomplete';
+import Autocomplete from '@mui/material/Autocomplete';
```

It requires `getOptionKey` if labels are not unique

`renderOption` should return `<li {...props}>`

Nested `<Chip>` doesn't require `key` property, because it's already returned by `getTagProps()`

5) `<TextField>` and `<Select>` are outlined by default in MUI 5, so added the following to the theme:

```
{
  components: {
    MuiTextField: {
      defaultProps: {
        variant: 'standard',
      },
    },
    MuiSelect: {
      defaultProps: {
        variant: 'standard',
      },
    },
  },
},
```

6) `makeStyles` is replaced by TSS version

https://mui.com/material-ui/migration/migrating-from-jss/

```diff
-import { makeStyles } from '@material-ui/core/styles';
+import { makeStyles } from 'tss-react/mui';
```

Also it doesn't support functions for each individual style property. Only a single function is supported

I didn't use the following version, because it doesn't work:
```
import { makeStyles } from '@mui/styles';
```

I guess that now it's recommended to use `styled` instead of `makeStyles`. But it's your decision, so I left `makeStyles` as is with slight syntax changes.

When we migrated our own app to MUI 5 we decided to drop CSS-in-JS completely, because:

- Modern CSS is just fine, it supports everything required, and there is no need for an additional abstraction level
- You have to learn CSS-in-JS, but the amount of documentation is way lower in comparison to plain CSS
- Plain CSS is much easier to write, maintain and understand. It's impossible to use some complicated functions in CSS
- When you separate TypeScript code and CSS styles in different files it's easier to understand and maintain the app
- CSS is a standard, it's backward compatible. You don't have to rewrite all styles after each library update. Who knows, maybe authors of React and MUI are planing to change CSS-in-JS library after each major release and maybe ten years later they will back to plain CSS :)
- We use CSS-classes in Cypress-tests instead of data-testid, ARIA attributes or explicit CSS. For instance `integration-tests/cypress/e2e/project/diagrams/collapse.cy.ts` at check `.should('have.css', 'border-bottom', '1px solid rgb(51, 176, 195)')` is failed on my machine, because the border width is `0.8px`. So I prefer to check existence of a CSS-class instead

But it's just our private opinion on CSS-in-JS, not absolute truth

7) I had to increase specificity of some styles because they were ignored:

```diff
-select: {
-  display: 'flex',
-  alignItems: 'center',
-},
+select: {
+  '&': {
+    display: 'flex',
+    alignItems: 'center',
+  },
+},
```

For example in `packages/forms/frontend/sirius-components-widget-reference/src/modals/CreateModal.tsx`

8) `nth-child(2)` in `makeStyles` caused runtime warnings so I replaced it by a separate CSS-class in `packages/core/frontend/sirius-components-core/src/workbench/RepresentationNavigation.tsx`

9) I had to add the following line at the beginning of `packages/core/frontend/sirius-components-core/src/index.ts` to fix the error "styled_default is not a function":

```
import '@mui/material/styles/styled';
```

See for details https://github.com/vitejs/vite/issues/12423#issuecomment-2080351394

10) `<Select>` role attribute is changed from `button` to `combobox` in MUI . So I updated Cypress tests

11) `<Tooltip>` adds `aria-label` instead of `title` to a nested `<Link>`

12) Tooltips works for disabled buttons only if the later one is surrounded by span

https://mui.com/material-ui/react-tooltip/#disabled-elements

```diff
<Tooltip title="Arrange all elements">
+ <span>
    <IconButton
      disabled={readOnly}>
      <AccountTreeIcon />
    </IconButton>
+ </span>
</Tooltip>
```

13) `theme.spacing(1)` returns string `8px` instead of number `8`. So I had to add `parseInt()` in `packages/portals/frontend/sirius-components-portals/src/representations/PortalRepresentation.tsx`

14) `theme.palette.text.hint` is removed https://github.com/mui/material-ui-pickers/issues/1681

15) Slight MUI 5 API changes:

```diff
-<Tabs scrollButtons="on" />
+<Tabs scrollButtons />
```

```diff
-<MenuItem button />
+<MenuItem />
```

16) It seems that `<Popper>` doesn't support `transition` attribute. If the attribute is specified then Popper is shown in the left upper corner.

Removed `trnasition` from `packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/tool-section/ToolSection.tsx` and `packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/group-tool/GroupPalette.tsx`

17) Added `@emotion/react` peer dependency to `sirius-components-forms` to fix "You are loading @emotion/react when it is already loaded. Running multiple instances may cause problems. This can happen if multiple versions are used, or if multiple builds of the same version are used."

# Cypress tests

I've fixed most of the tests after migration. But there are still some problems. Tests work unstable on my machine. Maybe it works better in your environment. Could you please help to fix it?

Some tests are failed when Cypress is run in a headless mode, probably because of problems with a clipboard or something else.

## When run in an interactive mode the following tests are failed for Chrome:

1) `integration-tests/cypress/e2e/project/diagrams/collapse.cy.ts`

expected '<div.nopan>' to have CSS property 'border-bottom' with the value '1px solid rgb(51, 176, 195)', but the value was '0.8px solid rgb(51, 176, 195)'

I tried to change Zoom factor in a browser, but it doesn't help. I think the test should not rely on a particular border width.

## For Electron the following tests are failed:

1) `integration-tests/cypress/e2e/project/diagrams/collapse.cy.ts`

In file `integration-tests/cypress/workbench/Explorer.ts` the test is failed on `cy.getByTestId('new-object-modal').should('not.exist')`. But if I add `cy.wait(2000)` before this line, the test is passed:

```
  public createObject(objectTreeItemLabel: string, childCreationDescriptionLabel: string) {
    this.getTreeItemByLabel(objectTreeItemLabel).find('button').click();
    cy.getByTestId('new-object').click();

    cy.getByTestId('childCreationDescription').children('[role="combobox"]').invoke('text').should('have.length.gt', 1);
    cy.getByTestId('childCreationDescription').click();
    cy.getByTestId('childCreationDescription')
      .get(`[data-value="${childCreationDescriptionLabel}"]`)
      .should('exist')
      .click();
    cy.getByTestId('create-object').click();
    // The test is fixed if I add the following line:
    // cy.wait(2000);
    cy.getByTestId('new-object-modal').should('not.exist');
  }
```

2) `integration-tests/cypress/e2e/project/diagrams/group-palette.cy.ts`

Error: "expected [data-testid="GroupPalette"] to exist in the DOM"

3) `integration-tests/cypress/e2e/project/diagrams/node-aspect-ratio.cy.ts`

Error: "Expected to find element: [data-testid="option-Node"], but never found it."

# Vitest

React component snapshots are outdated after migration. And to be honest I have no idea how to fix it

### PR DESCRIPTION
**Pull request by AresEkb**
# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

**Comment by AresEkb** [comment]
Two Cypress-tests are unstable:
- `cypress/e2e/project/forms/widget-reference.cy.ts`
- `cypress/e2e/project/portals/portals.cy.ts`

They sometimes fail. But I can't reproduce it locally.

**Comment by gcoutable** [comment]
There are still references to `material-ui@v4` in `package-lock.json`.
Can you generate `package-lock.json` from scratch?

**Comment by AresEkb** [comment]
> There are still references to `material-ui@v4` in `package-lock.json`. Can you generate `package-lock.json` from scratch?

I've regenerated. But there are still Material UI 4 dependencies because of gantt-task-react

Created a pull request to migrate it to MUI 5 https://github.com/ObeoNetwork/gantt-task-react/pull/9

Probably it can be made in 3 stages:
1) Migrate Sirius Web to MUI 5
2) Migrate gantt-task-react to MUI 5
3) Migrate Sirius Web to new gantt-task-react

Or in two stages:
1) Migrate gantt-task-react to MUI 5
2) Migrate Sirius Web to MUI 5 and new gantt-task-react

**Comment by gcoutable** [comment]
> I've regenerated. But there are still Material UI 4 dependencies because of gantt-task-react
> 
> Created a pull request to migrate it to MUI 5 [ObeoNetwork/gantt-task-react#9](https://github.com/ObeoNetwork/gantt-task-react/pull/9)
> 
> Probably it can be made in 3 stages:
> 
>     1. Migrate Sirius Web to MUI 5
> 
>     2. Migrate gantt-task-react to MUI 5
> 
>     3. Migrate Sirius Web to new gantt-task-react
> 
> 
> Or in two stages:
> 
>     1. Migrate gantt-task-react to MUI 5
> 
>     2. Migrate Sirius Web to MUI 5 and new gantt-task-react

We will do it in two stages.

Thank you for your contribution to gantt-task-react, I will look at it when I will have the time.

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/package.json
Comment (mcharfadi) [inline packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/package.json:1]: The list of widgets on the left of a FormDesciptionEditor representation contains icons with the wrong color.
image

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/ButtonWidget.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
Comment (mcharfadi) [inline packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/ButtonWidget.tsx:1]: The widget button has a different style (also in modals for creating representation where it's more noticeable)
After
image
Before
image

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/CheckboxWidget.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
Comment (mcharfadi) [inline packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/CheckboxWidget.tsx:1]: The checkbox and radio widget are not correctly align 
image
This is especially visible on the detail view
image

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/ImageWidget.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
Comment (mcharfadi) [inline packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/ImageWidget.tsx:1]: The image preview does to have the same style
image

Before :
image

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/RadioWidget.tsx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
Comment (mcharfadi) [inline packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/RadioWidget.tsx:1]: Same issue as the checkbox widget

File: packages/forms/frontend/sirius-components-forms/package.json [hunk 5]
```diff
@@ -35,10 +35,10 @@
     "@apollo/client": "3.10.4",
     "@eclipse-sirius/sirius-components-core": "*",
     "@eclipse-sirius/sirius-components-charts": "*",
+    "@emotion/react": "11.11.4",
```
Comment (mcharfadi) [inline packages/forms/frontend/sirius-components-forms/package.json]: Why the added dependency `"@emotion/react": "11.11.4",` here ?

File: packages/forms/frontend/sirius-components-forms/src/pagelist/PageList.tsx [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
Comment (mcharfadi) [inline packages/forms/frontend/sirius-components-forms/src/pagelist/PageList.tsx:1]: The items on the tabs are not spaced enough now
image
Before :
image

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/TreeWidget.tsx [hunk 7]
```diff
@@ -12,17 +12,17 @@
  *******************************************************************************/
```
Comment (mcharfadi) [inline packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/TreeWidget.tsx:12]: The checkbox of the Tree widget lost its style and also each treeItems is now too large
After
image
Before
image

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/RichTextWidget.tsx [hunk 8]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
Comment (mcharfadi) [inline packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/RichTextWidget.tsx:1]: We should probably increase the spacing between the buttons of the RichText widget
After
image
Before
image

File: packages/forms/frontend/sirius-components-widget-reference/src/ReferencePropertySection.tsx [hunk 9]
```diff
@@ -25,8 +25,8 @@ import {
   PropertySectionLabel,
```
Comment (mcharfadi) [inline packages/forms/frontend/sirius-components-widget-reference/src/ReferencePropertySection.tsx:25]: The ReferenceWidget buttons on the far right are too close to each others
image
Also when a value is set the background of the value is now transparent and too small
image
Before :
image

File: packages/sirius-web/frontend/sirius-web-application/src/diagrams/DiagramFilter.tsx [hunk 10]
```diff
@@ -11,18 +11,18 @@
  *     Obeo - initial API and implementation
```
Comment (mcharfadi) [inline packages/sirius-web/frontend/sirius-web-application/src/diagrams/DiagramFilter.tsx:11]: The filter elements modal (accessible from the last button of the diagram panel) is not positioned under the button anymore
image

File: packages/sirius-web/frontend/sirius-web-application/src/views/project-browser/list-projects-area/ListProjectsArea.tsx [hunk 11]
```diff
@@ -11,15 +11,15 @@
  *     Obeo - initial API and implementation
```
Comment (mcharfadi) [inline packages/sirius-web/frontend/sirius-web-application/src/views/project-browser/list-projects-area/ListProjectsArea.tsx:11]: The name of a project is always underlined on the list. Whereas before it was underlined if it was hovered
image

File: packages/sirius-web/frontend/sirius-web-application/src/views/edit-project/workbench-views/RelatedElementsView.tsx [hunk 12]
```diff
@@ -12,13 +12,13 @@
  *******************************************************************************/
 import { useSelection, WorkbenchViewComponentProps } from '@eclipse-sirius/sirius-components-core';
```
Comment (mcharfadi) [inline packages/sirius-web/frontend/sirius-web-application/src/views/edit-project/workbench-views/RelatedElementsView.tsx:13]: The style of the buttons on the top of the Related views does not make it clear what is the state of the button
image

File: packages/validation/frontend/sirius-components-validation/src/ValidationView.tsx [hunk 13]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
Comment (mcharfadi) [inline packages/validation/frontend/sirius-components-validation/src/ValidationView.tsx:1]: ValidationView
I got a weird behavior when opening this view

**Review by mcharfadi — DISMISSED** [review]
I listed some issues that would need to be addressed before merging.
I'm about to review https://github.com/ObeoNetwork/gantt-task-react/pull/9 and then I will try to fixe the above.

**Comment by AresEkb** [comment]
During one of rebases I forgot to update dependencies from MUI 4 to 5 in TreeItemAction.tsx. That was the reason of most style problems. Conflicts between MUI 4 and 5 can cause such a problems. It should be fine now

I've also fixed menu position, project link underline and removed dependency on @emotion/react

**Comment by mcharfadi** [comment]
Thanks you @AresEkb, this indeed fixed most issues.
I'll release a new version of [gantt-task-react](https://github.com/ObeoNetwork/gantt-task-react) now before doing what should be the last review.

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/TreeWidget.tsx (hunk 7 above)
Comment (mcharfadi) [reply packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/TreeWidget.tsx:12]: image
Another example

File: packages/forms/frontend/sirius-components-widget-reference/src/ReferencePropertySection.tsx (hunk 9 above)
Comment (mcharfadi) [reply packages/forms/frontend/sirius-components-widget-reference/src/ReferencePropertySection.tsx:25]: The button are now correctly placed but they have the wrong color
image

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/TreeWidget.tsx (hunk 7 above)
Comment (mcharfadi) [reply packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/TreeWidget.tsx:12]: Seems to be the padding of MuiTreeItem-content

File: packages/deck/frontend/sirius-components-deck/package.json [hunk 14]
```diff
@@ -33,8 +33,8 @@
     "@apollo/client": "3.10.4",
     "@eclipse-sirius/sirius-components-core": "*",
     "@ObeoNetwork/react-trello": "2.4.11",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
+    "@mui/icons-material": "5.15.19",
```
Comment (sbegaudeau) [inline packages/deck/frontend/sirius-components-deck/package.json:36]: The new peer-dependency `tss-react` is missing here (and I suspect in other package.json files). It's also missing in the devDependencies part.

File: packages/diagrams/frontend/sirius-components-diagrams/package.json [hunk 15]
```diff
@@ -32,8 +32,8 @@
   "peerDependencies": {
     "@apollo/client": "3.10.4",
     "@eclipse-sirius/sirius-components-core": "*",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
+    "@mui/material": "5.15.19",
```
Comment (sbegaudeau) [inline packages/diagrams/frontend/sirius-components-diagrams/package.json:35]: peerDependencies and devDependencies to update

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/package.json [hunk 16]
```diff
@@ -36,8 +36,9 @@
     "@eclipse-sirius/sirius-components-core": "*",
     "@eclipse-sirius/sirius-components-charts": "*",
     "@eclipse-sirius/sirius-components-forms": "*",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
+    "@mui/icons-material": "5.15.19",
+    "@mui/material": "5.15.19",
+    "@mui/x-tree-view": "7.6.1",
```
Comment (sbegaudeau) [inline packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/package.json:41]: peer and dev dependencies to be updated

File: packages/forms/frontend/sirius-components-forms/package.json [hunk 17]
```diff
@@ -36,9 +36,8 @@
     "@eclipse-sirius/sirius-components-core": "*",
     "@eclipse-sirius/sirius-components-charts": "*",
     "@lexical/react": "0.8.1",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
-    "@material-ui/lab": "4.0.0-alpha.61",
+    "@mui/material": "5.15.19",
```
Comment (sbegaudeau) [inline packages/forms/frontend/sirius-components-forms/package.json:39]: peer/dev dependencies

File: packages/forms/frontend/sirius-components-forms/src/propertysections/SliderPropertySection.tsx [hunk 18]
```diff
@@ -10,9 +10,10 @@
  * Contributors:
  *     Obeo - initial API and implementation
  *******************************************************************************/
-import { gql, useMutation } from '@apollo/client';
+import { useMutation } from '@apollo/client';
 import { useMultiToast } from '@eclipse-sirius/sirius-components-core';
-import Slider from '@material-ui/core/Slider';
+import Slider from '@mui/material/Slider';
+import gql from 'graphql-tag';
```
Comment (sbegaudeau) [inline packages/forms/frontend/sirius-components-forms/src/propertysections/SliderPropertySection.tsx]: gql should be imported from Apollo client, we should probably ban import from graphql-tag

File: packages/forms/frontend/sirius-components-widget-reference/package.json [hunk 19]
```diff
@@ -37,9 +37,8 @@
     "@eclipse-sirius/sirius-components-forms": "*",
     "@eclipse-sirius/sirius-components-trees": "*",
     "@lexical/react": "0.8.1",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
-    "@material-ui/lab": "4.0.0-alpha.61",
+    "@mui/material": "5.15.19",
+    "@mui/icons-material": "5.15.19",
```
Comment (sbegaudeau) [inline packages/forms/frontend/sirius-components-widget-reference/package.json:41]: peer/dev dependencies to update with tss-react

File: packages/gantt/frontend/sirius-components-gantt/package.json [hunk 20]
```diff
@@ -31,10 +31,10 @@
   },
   "peerDependencies": {
     "@apollo/client": "3.10.4",
-    "@ObeoNetwork/gantt-task-react": "0.4.19",
+    "@ObeoNetwork/gantt-task-react": "0.5.0",
     "@eclipse-sirius/sirius-components-core": "*",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
+    "@mui/material": "5.15.19",
+    "@mui/icons-material": "5.15.19",
```
Comment (sbegaudeau) [inline packages/gantt/frontend/sirius-components-gantt/package.json:37]: peer/dev dependencies to update with tss-react

File: packages/portals/frontend/sirius-components-portals/package.json [hunk 21]
```diff
@@ -32,9 +32,8 @@
   "peerDependencies": {
     "@apollo/client": "3.10.4",
     "@eclipse-sirius/sirius-components-core": "*",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
-    "@material-ui/lab": "4.0.0-alpha.61",
+    "@mui/material": "5.15.19",
+    "@mui/icons-material": "5.15.19",
```
Comment (sbegaudeau) [inline packages/portals/frontend/sirius-components-portals/package.json:36]: peer/dev dependencies to update with tss-react

File: packages/selection/frontend/sirius-components-selection/package.json [hunk 22]
```diff
@@ -33,8 +33,8 @@
     "@apollo/client": "3.10.4",
     "@eclipse-sirius/sirius-components-core": "*",
     "@eclipse-sirius/sirius-components-diagrams": "*",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
+    "@mui/material": "5.15.19",
+    "@mui/icons-material": "5.15.19",
```
Comment (sbegaudeau) [inline packages/selection/frontend/sirius-components-selection/package.json:37]: Same as other package.json

File: packages/sirius-web/frontend/sirius-web-application/package.json [hunk 23]
```diff
@@ -53,17 +52,17 @@
     "react-dom": "17.0.2",
     "react-router-dom": "5.2.0",
     "reactflow": "11.10.1",
+    "tss-react": "4.9.7",
     "xstate": "4.32.1"
   },
   "devDependencies": {
     "@apollo/client": "3.10.4",
     "@eclipse-sirius/sirius-components-core": "*",
+    "@eclipse-sirius/sirius-components-deck": "*",
```
Comment (sbegaudeau) [inline packages/sirius-web/frontend/sirius-web-application/package.json:61]: tss-react is missing in the dev dependencies

File: packages/sirius-web/frontend/sirius-web-papaya/package.json [hunk 24]
```diff
@@ -41,9 +41,8 @@
     "@eclipse-sirius/sirius-components-trees": "*",
     "@eclipse-sirius/sirius-components-widget-reference": "*",
     "@eclipse-sirius/sirius-web-application": "*",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
-    "@material-ui/lab": "4.0.0-alpha.61",
+    "@mui/material": "5.15.19",
```
Comment (sbegaudeau) [inline packages/sirius-web/frontend/sirius-web-papaya/package.json:44]: tss-react missing in peer dependencies and mui and tss-react should also be in dev dependencies

File: packages/sirius-web/frontend/sirius-web/package.json [hunk 25]
```diff
@@ -27,10 +27,9 @@
     "@eclipse-sirius/sirius-components-validation": "*",
     "@eclipse-sirius/sirius-web-application": "*",
     "@eclipse-sirius/sirius-web-papaya": "*",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
-    "@material-ui/lab": "4.0.0-alpha.61",
-    "@ObeoNetwork/gantt-task-react": "0.4.19",
+    "@mui/icons-material": "5.15.19",
```
Comment (sbegaudeau) [inline packages/sirius-web/frontend/sirius-web/package.json:30]: tss-react should be there too

File: packages/trees/frontend/sirius-components-trees/package.json [hunk 26]
```diff
@@ -34,8 +34,8 @@
   "peerDependencies": {
     "@apollo/client": "3.10.4",
     "@eclipse-sirius/sirius-components-core": "*",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
+    "@mui/material": "5.15.19",
+    "@mui/icons-material": "5.15.19",
```
Comment (sbegaudeau) [inline packages/trees/frontend/sirius-components-trees/package.json:38]: tss-react

File: packages/validation/frontend/sirius-components-validation/package.json [hunk 27]
```diff
@@ -32,8 +32,8 @@
   "peerDependencies": {
     "@apollo/client": "3.10.4",
     "@eclipse-sirius/sirius-components-core": "*",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
+    "@mui/material": "5.15.19",
+    "@mui/icons-material": "5.15.19",
```
Comment (sbegaudeau) [inline packages/validation/frontend/sirius-components-validation/package.json:36]: tss-react in peer/dev dependencies

File: vscode-extension/package.json [hunk 28]
```diff
@@ -135,10 +135,9 @@
     "@eclipse-sirius/sirius-components-selection": "*",
     "@eclipse-sirius/sirius-components-trees": "*",
     "@eclipse-sirius/sirius-components-validation": "*",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
-    "@material-ui/lab": "4.0.0-alpha.61",
-    "@ObeoNetwork/gantt-task-react": "0.4.19",
+    "@mui/icons-material": "5.15.19",
```
Comment (sbegaudeau) [inline vscode-extension/package.json:138]: tss-react is missing here

File: packages/core/frontend/sirius-components-core/package.json [hunk 29]
```diff
@@ -33,18 +33,20 @@
...
   "peerDependencies": {
     "@apollo/client": "3.10.4",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
+    "@mui/icons-material": "5.15.19",
+    "@mui/material": "5.15.19",
     "@xstate/react": "1.6.3",
     "graphql": "16.8.1",
     "react": "17.0.2",
+    "react-dom": "17.0.2",
```
Comment (sbegaudeau) [inline packages/core/frontend/sirius-components-core/package.json:41]: Is react-dom really necessary here? Unless I missed something it should be removed

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
I'll take over this PR to fix the last issue that I have and ensure that it gets merged as soon as possible

File: packages/forms/frontend/sirius-components-forms/src/propertysections/SliderPropertySection.tsx (hunk 18 above)
Comment (sbegaudeau) [reply packages/forms/frontend/sirius-components-forms/src/propertysections/SliderPropertySection.tsx]: Done

File: packages/core/frontend/sirius-components-core/package.json (hunk 29 above)
Comment (sbegaudeau) [reply packages/core/frontend/sirius-components-core/package.json:41]: After analysis, it's the other way around, `react-dom` is a peer dependency of `@mui/material` so it should be everywhere (which is currently not the case).

File: packages/deck/frontend/sirius-components-deck/package.json (hunk 14 above)
Comment (sbegaudeau) [reply packages/deck/frontend/sirius-components-deck/package.json:36]: Done

File: packages/diagrams/frontend/sirius-components-diagrams/package.json (hunk 15 above)
Comment (sbegaudeau) [reply packages/diagrams/frontend/sirius-components-diagrams/package.json:35]: Done

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/package.json (hunk 16 above)
Comment (sbegaudeau) [reply packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/package.json:41]: Done

File: packages/forms/frontend/sirius-components-forms/package.json (hunk 17 above)
Comment (sbegaudeau) [reply packages/forms/frontend/sirius-components-forms/package.json:39]: Done

File: packages/forms/frontend/sirius-components-widget-reference/package.json (hunk 19 above)
Comment (sbegaudeau) [reply packages/forms/frontend/sirius-components-widget-reference/package.json:41]: Done

File: packages/gantt/frontend/sirius-components-gantt/package.json (hunk 20 above)
Comment (sbegaudeau) [reply packages/gantt/frontend/sirius-components-gantt/package.json:37]: Done

File: packages/portals/frontend/sirius-components-portals/package.json (hunk 21 above)
Comment (sbegaudeau) [reply packages/portals/frontend/sirius-components-portals/package.json:36]: Done

File: packages/selection/frontend/sirius-components-selection/package.json (hunk 22 above)
Comment (sbegaudeau) [reply packages/selection/frontend/sirius-components-selection/package.json:37]: Done

File: packages/sirius-web/frontend/sirius-web-application/package.json (hunk 23 above)
Comment (sbegaudeau) [reply packages/sirius-web/frontend/sirius-web-application/package.json:61]: Done

File: packages/sirius-web/frontend/sirius-web-papaya/package.json (hunk 24 above)
Comment (sbegaudeau) [reply packages/sirius-web/frontend/sirius-web-papaya/package.json:44]: Done

File: packages/sirius-web/frontend/sirius-web/package.json (hunk 25 above)
Comment (sbegaudeau) [reply packages/sirius-web/frontend/sirius-web/package.json:30]: Done

File: packages/trees/frontend/sirius-components-trees/package.json (hunk 26 above)
Comment (sbegaudeau) [reply packages/trees/frontend/sirius-components-trees/package.json:38]: Done

File: packages/validation/frontend/sirius-components-validation/package.json (hunk 27 above)
Comment (sbegaudeau) [reply packages/validation/frontend/sirius-components-validation/package.json:36]: Done

File: vscode-extension/package.json (hunk 28 above)
Comment (sbegaudeau) [reply vscode-extension/package.json:138]: Done

File: packages/core/frontend/sirius-components-core/package.json (hunk 29 above)
Comment (sbegaudeau) [reply packages/core/frontend/sirius-components-core/package.json:41]: Done

**Review by sbegaudeau — APPROVED** [review]

**Comment by sbegaudeau** [comment]
Thanks Denis for the contribution, it took quite some time to integrate it in Sirius Web but it's now finally merged \o/

**Comment by AresEkb** [comment]
Thanks a lot for merging!

**Pull request merged**